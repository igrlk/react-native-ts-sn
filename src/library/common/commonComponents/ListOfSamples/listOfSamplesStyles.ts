import EStyleSheet from 'react-native-extended-stylesheet';

export default EStyleSheet.create({
	sample: {
		flexDirection: 'row',
		justifyContent: 'space-between',
		alignItems: 'center',
		paddingBottom: 7,
		marginBottom: 10,
		borderBottomWidth: 1,
		borderColor: '#E1E1E1',
	},
	sampleId: {
		fontFamily: 'Poppins',
		fontSize: '$fontSize.small',
		marginBottom: 8,
		color: '$colorBlack',
	},
	sampleText: {
		flexDirection: 'row',
		justifyContent: 'space-between',
		alignItems: 'center',
	},
	sampleTextValue: {
		fontFamily: 'Poppins',
		fontSize: '$fontSize.xmini',
		color: '#9E9E9E',
	},
	circle: {
		width: 8,
		height: 8,
		backgroundColor: '#EA4640',
		marginRight: 6,
		borderRadius: 100,
	},
});
