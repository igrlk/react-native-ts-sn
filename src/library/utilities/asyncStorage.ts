import { AsyncStorage } from 'react-native';

export const prefix = 'EventNetwork/';

export default class Storage {
	static async getItem(name: string) {
		const item = await AsyncStorage.getItem(prefix + name);

		if (item) return JSON.parse(item);

		return null;
	}

	static setItem(name: string, value: any) {
		return AsyncStorage.setItem(prefix + name, JSON.stringify(value));
	}

	static removeItem(name: string) {
		return AsyncStorage.removeItem(prefix + name);
	}
}
