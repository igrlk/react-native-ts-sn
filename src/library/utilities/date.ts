import { format } from 'date-fns';

export function getDateForSample(miliseconds: number | string) {
  return format(new Date(+miliseconds), 'MMM  d, yyyy');
}
