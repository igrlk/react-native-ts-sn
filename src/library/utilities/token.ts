import AsyncStorage from 'library/utilities/asyncStorage';

import { resetNavigation, navigateTo } from './navigator';

export const name = 'authToken';

export const setToken = (authToken: string) => AsyncStorage.setItem(name, authToken);

export const getToken = async () => {
	const token = await AsyncStorage.getItem(name);
	if (await AsyncStorage.getItem('isTokenForRegistration')) {
		AsyncStorage.removeItem('isTokenForRegistration');
		AsyncStorage.removeItem(name);
	}

	return token;
};

export const checkTokenForLogin = async () => {
	const authToken = await getToken();

	if (authToken) {
		resetNavigation('Landing');

		return true;
	}

	return false;
};

export const removeToken = async () => AsyncStorage.removeItem(name);

export const loginUser = async ({
	authToken,
	email,
	uuid,
	showOnboarding,
}: {
	authToken: string;
	email: string;
	uuid: string;
	showOnboarding?: boolean;
}) => {
	await Promise.all([
		AsyncStorage.setItem(name, authToken),
		AsyncStorage.setItem('email', email),
		AsyncStorage.setItem('uuid', uuid),
	]);
	if (showOnboarding) {
		resetNavigation('Onboarding');
	} else {
		resetNavigation('Landing');
	}
};

export const logout = () => {
	AsyncStorage.removeItem(name);
	resetNavigation('Login');
};
