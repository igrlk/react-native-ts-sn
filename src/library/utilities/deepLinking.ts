import { LinkingStatic } from 'react-native';

import { navigateTo } from 'library/utilities/navigator';

export type DeepLinkingUrl = string | { url: string } | null;

export function handleDeepLinking(deepLinkingUrl: DeepLinkingUrl) {
  const url = getUrl(deepLinkingUrl);
  console.log(url);
  if (!url) return;

  if (url.indexOf('signup/') === 0) {
    navigateTo('SignUpAddPassword', getParamsForSignUp(url));
  }
}

export function getUrl(deepLinkingUrl: DeepLinkingUrl) {
  const preprocessedUrl = deepLinkingUrl
    ? typeof deepLinkingUrl === 'string'
      ? deepLinkingUrl
      : deepLinkingUrl.url
    : null;

  if (!preprocessedUrl) return;

  return preprocessedUrl.slice(14);
}

export function getParamsForSignUp(url: string) {
  const [signUpEmail, signUpKey] = url.slice(7).split('/');

  return { signUpKey, signUpEmail };
}

export function bindDeepLinkingHandlers(Linking: LinkingStatic) {
  return () => {
    Linking.getInitialURL().then(handleDeepLinking);
    Linking.addEventListener('url', handleDeepLinking);
  };
}
