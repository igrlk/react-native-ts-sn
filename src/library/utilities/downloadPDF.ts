import RNHTMLtoPDF from 'react-native-html-to-pdf';

import { navigateTo } from 'library/utilities/navigator';

export default function dowloadPdf(options: any, screenName: string, backToScreen: string) {
    RNHTMLtoPDF.convert(options).then((file: any) => {
        navigateTo(screenName, { backToScreen });
    })
    .catch((err: Error) => {});
}