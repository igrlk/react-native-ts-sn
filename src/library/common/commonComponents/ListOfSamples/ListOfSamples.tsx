import React from 'react';
import { View, Text, TouchableOpacity } from 'react-native';

import ArrowRight from 'resources/images/svg/ArrowRight';
import { getNavigateTo } from 'library/utilities/navigator';
import { getDateForSample } from 'library/utilities/date';
import getStatusColor from 'library/utilities/getStatusColor';

import styles from './listOfSamplesStyles';

interface ListOfSamplesProps {
	samples: any[];
}

export default function ListOfSamples({ samples }: ListOfSamplesProps) {
	return (
		<View>
			{samples
				.sort((a, b) => b.created - a.created)
				.map(sample => (
					<TouchableOpacity
						key={sample.uuid}
						onPress={getNavigateTo('OrderDetail', { sample })}
						style={styles.sample}
					>
						<View>
							<Text style={styles.sampleId}>{getNumberOfSample(sample)}</Text>
							<View style={styles.sampleText}>
								<View style={[styles.circle, getStatusColor(sample.status)]} />
								<Text style={styles.sampleTextValue}>
									{sample.status} ({getDateForSample(sample.created)})
								</Text>
							</View>
						</View>
						<ArrowRight />
					</TouchableOpacity>
				))}
		</View>
	);
}

export const getNumberOfSample = (sample : any) => {
	return sample.number ? sample.number : 'New Request';
}
