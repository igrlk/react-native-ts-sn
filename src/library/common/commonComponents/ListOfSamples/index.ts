import ListOfSamples from './ListOfSamples';

export default ListOfSamples;
