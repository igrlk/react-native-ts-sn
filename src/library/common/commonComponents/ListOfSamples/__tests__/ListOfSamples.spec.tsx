import React from 'react';
import renderer from 'react-test-renderer';
import ApolloClient from 'apollo-boost';
import { ApolloProvider } from '@apollo/react-hooks';

import { getToken } from 'library/utilities/token';
import ListOfSamples from '../ListOfSamples';
import * as ListOfSamplesHandlers from '../ListOfSamples';

const client = new ApolloClient({
	uri: 'http://sentinel.oioio.ru:8889/graphql',
	fetch: jest.fn(),
	request: async operation => {
		const authToken = await getToken();
		operation.setContext({
			headers: {
				authorization: authToken ? `Bearer ${authToken}` : '',
			},
		});
	},
});

describe('ListOfSamples', () => {
	it('match to snapshot', () => {
        const samples = [{
            status: 'Completed',
            number: null,
            created: '1112222121',
        }];
		const tree = renderer.create(
        <ApolloProvider client={client}>
            <ListOfSamples samples={samples} />
        </ApolloProvider>,).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('check call function getNumberOfSample with New Request sample', () => {
        jest.spyOn(ListOfSamplesHandlers, 'getNumberOfSample');
        const sample = {
            uuid: 1,
            status: 'Completed',
            number: null,
            created: '1112222121',
        };

		ListOfSamplesHandlers.getNumberOfSample(sample);
		expect(ListOfSamplesHandlers.getNumberOfSample).toReturnWith('New Request')
    });
    
    it('check call function getNumberOfSample', () => {
        jest.spyOn(ListOfSamplesHandlers, 'getNumberOfSample');
        const sample = {
            uuid: 1,
            status: 'Completed',
            number: 702,
            created: '1112222121',
        };

		ListOfSamplesHandlers.getNumberOfSample(sample);
		expect(ListOfSamplesHandlers.getNumberOfSample).toReturnWith(702)
	});
});