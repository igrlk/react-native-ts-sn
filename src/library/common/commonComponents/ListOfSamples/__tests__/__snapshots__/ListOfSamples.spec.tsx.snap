// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListOfSamples match to snapshot 1`] = `
<View>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View>
      <Text>
        New Request
      </Text>
      <View>
        <View
          style={
            Array [
              undefined,
              Object {
                "backgroundColor": "#29D98E",
              },
            ]
          }
        />
        <Text>
          Completed
           (
          Jan  13, 1970
          )
        </Text>
      </View>
    </View>
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="15"
      bbWidth="9"
      fill="none"
      focusable={false}
      height="15"
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          undefined,
          Object {
            "opacity": 1,
          },
          Object {
            "flex": 0,
            "height": 15,
            "width": 9,
          },
        ]
      }
      vbHeight={15}
      vbWidth={9}
      width="9"
    >
      <RNSVGGroup
        fill={null}
        fillOpacity={1}
        fillRule={1}
        font={Object {}}
        matrix={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
        opacity={1}
        originX={0}
        originY={0}
        propList={
          Array [
            "fill",
          ]
        }
        rotation={0}
        scaleX={1}
        scaleY={1}
        skewX={0}
        skewY={0}
        stroke={null}
        strokeDasharray={null}
        strokeDashoffset={null}
        strokeLinecap={0}
        strokeLinejoin={0}
        strokeMiterlimit={4}
        strokeOpacity={1}
        strokeWidth={1}
        vectorEffect={0}
        x={0}
        y={0}
      >
        <RNSVGPath
          d="M7.5981 7.50039L0.823859 14.2746C0.657254 14.4412 0.657254 14.7084 0.823859 14.875C0.990465 15.0417 1.25766 15.0417 1.42427 14.875L8.50029 7.79903C8.6669 7.63242 8.6669 7.36522 8.50029 7.19862L1.42427 0.12574C1.34254 0.0440092 1.23251 0 1.12563 0C1.01876 0 0.908733 0.0408659 0.827002 0.12574C0.660397 0.292345 0.660397 0.559543 0.827002 0.726148L7.5981 7.50039Z"
          fill={
            Array [
              0,
              4291085508,
            ]
          }
          fillOpacity={1}
          fillRule={1}
          matrix={
            Array [
              1,
              0,
              0,
              1,
              0,
              0,
            ]
          }
          opacity={1}
          originX={0}
          originY={0}
          propList={
            Array [
              "fill",
            ]
          }
          rotation={0}
          scaleX={1}
          scaleY={1}
          skewX={0}
          skewY={0}
          stroke={null}
          strokeDasharray={null}
          strokeDashoffset={null}
          strokeLinecap={0}
          strokeLinejoin={0}
          strokeMiterlimit={4}
          strokeOpacity={1}
          strokeWidth={1}
          vectorEffect={0}
          x={0}
          y={0}
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
</View>
`;
